### ISSUE TITLE
TypeError: r.matches is not a function

### ISSUE THREAD

**Issue by jiahouping** [carbon-design-system/carbon#6287]
Hi, sometimes, we will hit an error "TypeError: r.matches is not a function" as below:
<img width="1555" alt="Screen Shot 2020-06-16 at 3 12 04 PM" src="https://user-images.githubusercontent.com/20769520/84749439-8ebb3e80-afec-11ea-808a-c313a2176a03.png">

And we have investigated that this error happens in "carbon-components-react/es/components/OverflowMenu/OverflowMenu.js".
<img width="853" alt="Screen Shot 2020-06-16 at 4 18 50 PM" src="https://user-images.githubusercontent.com/20769520/84749895-2c167280-afed-11ea-9387-80906e400982.png">

This target is an HTMLDocument and it calls target.matches(here, matches is not defined) , it will throw such error.

<img width="672" alt="Screen Shot 2020-06-16 at 4 19 21 PM" src="https://user-images.githubusercontent.com/20769520/84750078-6b44c380-afed-11ea-8a56-864c5ea7002c.png">

Could you please help to take a look at this issue? Thanks.

Carbon version:
"carbon-components": "^10.8.1",
"carbon-components-react": "^7.10.2",

**Comment by emyarod** [comment]
can you create a [reduced test case in Code Sandbox](https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox) for easier testing and debugging?

**Comment by joshblack** [comment]
Hey there! 👋  Going to close this out after 14 days of inactivity. Feel free to comment and we can re-open!
